collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh

```diff
@@ -1,311 +1,528 @@
+  ulimit -aH > $TMPDIR/systeminfo/ulimit-hard 2>&1
+  uname -a > $TMPDIR/systeminfo/uname 2>&1
+  cat /etc/*release > $TMPDIR/systeminfo/osrelease 2>&1
+  if $(command -v lsblk >/dev/null 2>&1); then
+    lsblk > $TMPDIR/systeminfo/lsblk 2>&1
+  fi
+  lsof -Pn > $TMPDIR/systeminfo/lsof 2>&1 & timeout_cmd
+  if $(command -v sysctl >/dev/null 2>&1); then
+    sysctl -a > $TMPDIR/systeminfo/sysctla 2>/dev/null
+  fi
+  if $(command -v systemctl >/dev/null 2>&1); then
+    systemctl list-units > $TMPDIR/systeminfo/systemd-units 2>&1
+  fi
+  if $(command -v service >/dev/null 2>&1); then
+    service --status-all > $TMPDIR/systeminfo/service-statusall 2>&1
+  fi
 
-timeout_start_msg "lsof"
-lsof -Pn > $TMPDIR/systeminfo/lsof 2>&1 & timeout_cmd
-timeout_done_msg
+}
 
-if $(command -v sysctl >/dev/null 2>&1); then
-  sysctl -a > $TMPDIR/systeminfo/sysctla 2>/dev/null
-fi
-if $(command -v systemctl >/dev/null 2>&1); then
-  systemctl list-units > $TMPDIR/systeminfo/systemd-units 2>&1
-fi
-if $(command -v service >/dev/null 2>&1); then
-  service --status-all > $TMPDIR/systeminfo/service-statusall 2>&1
-fi
+system-ubuntu() {
 
-# OS: Ubuntu
-if $(command -v ufw >/dev/null 2>&1); then
-  ufw status > $TMPDIR/systeminfo/ubuntu-ufw 2>&1
-fi
-if $(command -v apparmor_status >/dev/null 2>&1); then
-  apparmor_status > $TMPDIR/systeminfo/ubuntu-apparmorstatus 2>&1
-fi
-if $(command -v dpkg >/dev/null 2>&1); then
-  dpkg -l > $TMPDIR/systeminfo/packages-dpkg 2>&1
-fi
+  if $(command -v ufw >/dev/null 2>&1); then
+    ufw status > $TMPDIR/systeminfo/ubuntu-ufw 2>&1
+  fi
+  if $(command -v apparmor_status >/dev/null 2>&1); then
+    apparmor_status > $TMPDIR/systeminfo/ubuntu-apparmorstatus 2>&1
+  fi
+  if $(command -v dpkg >/dev/null 2>&1); then
+    dpkg -l > $TMPDIR/systeminfo/packages-dpkg 2>&1
+  fi
+
+}
+
+system-rhel() {
 
-# OS: RHEL
-if [ -f /etc/redhat-release ]; then
   systemctl status NetworkManager > $TMPDIR/systeminfo/rhel-statusnetworkmanager 2>&1
   systemctl status firewalld > $TMPDIR/systeminfo/rhel-statusfirewalld 2>&1
   if $(command -v getenforce >/dev/null 2>&1); then
-  getenforce > $TMPDIR/systeminfo/rhel-getenforce 2>&1
+    getenforce > $TMPDIR/systeminfo/rhel-getenforce 2>&1
+  fi
+  if $(command -v rpm >/dev/null 2>&1); then
+    rpm -qa > $TMPDIR/systeminfo/packages-rpm 2>&1
   fi
-fi
-if $(command -v rpm >/dev/null 2>&1); then
-  rpm -qa > $TMPDIR/systeminfo/packages-rpm 2>&1
-fi
 
-# Docker
-echo "Collecting Docker info"
-mkdir -p $TMPDIR/docker
+}
+
+networking() {
+
+  techo "Collecting network output"
+  mkdir -p $TMPDIR/networking
+  iptables-save > $TMPDIR/networking/iptablessave 2>&1
+  iptables --wait 1 --numeric --verbose --list --table mangle > $TMPDIR/networking/iptablesmangle 2>&1
+  iptables --wait 1 --numeric --verbose --list --table nat > $TMPDIR/networking/iptablesnat 2>&1
+  iptables --wait 1 --numeric --verbose --list > $TMPDIR/networking/iptables 2>&1
```

Comment:
This failed with 
```
Bad argument `1'
Try `iptables -h' or 'iptables --help' for more information.
```
on SLES 12 SP2.
Other iptables commands failed as well with same error.

Reply:
The version of iptables in SLES 12 SP5 (`iptables v1.4.21`) doesn't support an argument (seconds) for `--wait`, we could remove the argument and just wait, or remove the `--wait` all together.

Reply:
Added condition for `sles` to avoid using the `--wait` flag